bluetooth: adv_prov: device_name: Add device name only in pairing mode (#13928)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

    Disabling this Kconfig option improves the debugging experience.
  * The MCUboot, B0, and HCI RPMsg child images release configurations to explicitly enable the :kconfig:option:`CONFIG_RESET_ON_FATAL_ERROR` Kconfig option.
    Enabling this Kconfig option improves the reliability of the firmware.
  * The default value of newly introduced :kconfig:option:`CONFIG_BT_ADV_PROV_DEVICE_NAME_PAIRING_MODE_ONLY` Kconfig option.
    The option is disabled by default by nRF Desktop application.

>     Bluetooth device name is provided in scan response also outside of pairing mode for backwards compatibility.

Reviewer: Maybe add ... in **the** scan response ...

  * Added the :c:func:`bt_scan_update_connect_if_match` function to update the autoconnect flag after a filter match.

> * :ref:`bt_le_adv_prov_readme`:
> 
>   * By default, the Bluetooth device name provider provides data only in the pairing mode (:c:member:`bt_le_adv_prov_adv_state.pairing_mode`).
>     The newly introduced Kconfig option (:kconfig:option:`CONFIG_BT_ADV_PROV_DEVICE_NAME_PAIRING_MODE_ONLY`) can be disabled to provide the device name also when the device is not in the pairing mode.
> 

Reviewer 2: Sorry, my bad, I completely misread the heading. As @peknis suggested, we should align this with the entire section

```suggestion
* :ref:`bt_le_adv_prov_readme`:

  * Added an option to provide the device name also when the device is not in the pairing mode by disabling the newly introduced Kconfig option (:kconfig:option:`CONFIG_BT_ADV_PROV_DEVICE_NAME_PAIRING_MODE_ONLY`).
  By default, the Bluetooth device name provider provides data only in the pairing mode
```
tbc if the meaning is correct

Author: To be honest, I would prefer to keep the information about modified default behavior as the first one. The Kconfig option was introduced to allow easily brining back the previous behavior if needed.

  * Added the :c:func:`bt_scan_update_connect_if_match` function to update the autoconnect flag after a filter match.

> * :ref:`bt_le_adv_prov_readme`:
> 
>   * Bluetooth device name provider by default provides data only when in pairing mode (:c:member:`bt_le_adv_prov_adv_state.pairing_mode`).
>     The newly introduced Kconfig option (:kconfig:option:`CONFIG_BT_ADV_PROV_DEVICE_NAME_PAIRING_MODE_ONLY`) can be disabled to provide device name also if not in pairing mode.

Reviewer 3: ```suggestion
* :ref:`bt_le_adv_prov_readme`:

  * Bluetooth device name provider by default provides data only when in pairing mode (:c:member:`bt_le_adv_prov_adv_state.pairing_mode`).
    The newly introduced Kconfig option (:kconfig:option:`CONFIG_BT_ADV_PROV_DEVICE_NAME_PAIRING_MODE_ONLY`) can be disabled to provide the device name also when the device is not in the pairing mode.
```

  * Added the :c:func:`bt_scan_update_connect_if_match` function to update the autoconnect flag after a filter match.

* :ref:`bt_le_adv_prov_readme`:

>   * Bluetooth device name provider by default provides data only when in pairing mode (:c:member:`bt_le_adv_prov_adv_state.pairing_mode`).

Author: I would prefer to use `provides` instead of `sends` here. The provider only passes the data to `bt_le_adv_prov` subsystem (then an application takes data from subsystem ands sends the data over radio).
